master/pkg/searcher/random.go

```diff
@@ -68,13 +68,24 @@ func (s *randomSearch) initialOperations(ctx context) ([]Operation, error) {
 	return ops, nil
 }
 
-func (s *randomSearch) progress(trialProgress map[model.RequestID]model.PartialUnits) float64 {
+func (s *randomSearch) progress(
+	trialProgress map[model.RequestID]model.PartialUnits,
+	trialsClosed map[model.RequestID]bool) float64 {
 	if s.MaxConcurrentTrials() > 0 && s.PendingTrials > s.MaxConcurrentTrials() {
 		panic("pending trials is greater than max_concurrent_trials")
 	}
-	unitsCompleted := sumTrialLengths(trialProgress)
+	unitsCompleted := 0.
+	for k, v := range trialProgress {
```

Comment:
this (and grid) is a bit tricky to understand it works properly, since it only because we delete the progress of invalid hp trials (it's not 'locally understandable'). maybe just drop a comment noting this?